src/ruby/yast/rspec/helpers.rb

```diff
@@ -0,0 +1,69 @@
+require "yast"
+
+module Yast
+  module RSpec
+    module Helpers
+      # Helper for defining YaST modules which might not be present in
+      # the system when running the tests.
+      #
+      # When the requested module is present in the system then it is imported
+      # as usually. If it is missing a substitute definition is created.
+      #
+      # @note This needs to be called *after* starting code coverage with
+      #   `SimpleCov.start` so the coverage of the imported modules is correctly
+      #   counted.
+      #
+      # @note You can force using the defined stubs although the modules are
+      #   present in the system, this might be useful to actually test the stubs.
+      #
+      # @example Mock the `Language` module and the `Language.language` method
+      # Yast::RSpec::Helpers.define_yast_module("Language") do
+      #   # see modules/Language.rb in yast2-country
+      #   module Yast
+      #     class LanguageClass < Module
+      #       # @return [String]
+      #       def language; end
+      #     end
+      #     Language = LanguageClass.new
+      #   end
+      # end
+      #
+      # @example Mock empty `AutoInstall` module
+      # Yast::RSpec::Helpers.define_yast_module("AutoInstall")
+      #
+      # @param name [String] name of the YaST module
+      # @param block [Block] optional module definition, it should provide
+      #   the same API as the original module, it is enough to define only the
+      #   methods used in the tests, if no block is passed an empty module is defined
+      def self.define_yast_module(name, &block)
+        # sanity check, make sure the code coverage is already running if it is enabled
+        if ENV["COVERAGE"] && (!defined?(SimpleCov) || !SimpleCov.running)
+          abort "\nERROR: The `define_yast_module` method needs to be called *after* " \
+            "enabling the code coverage tracking with `SimpleCov.start`!\n" \
+            "  Called from: #{caller(1).first}\n\n"
+        end
+
+        if ENV["YAST_FORCE_MODULE_STUBS"]
+          define_missing_yast_module(name, &block)
+        else
+          # try loading the module, it might be present in the system (running locally
+          # or in GitHub Actions), mock it only when missing (e.g. in OBS build)
+          Yast.import(name)
+          puts "Found module Yast::#{name}"
+        end
+      rescue NameError
+        define_missing_yast_module(name, &block)
+      end
+
+      private_class_method def self.define_missing_yast_module(name, &block)
+        warn "Module Yast::#{name} not found"
+        if block_given?
+          block.call
+        else
+          # create a fake implementation of the module
+          Yast.const_set(name.to_sym, Class.new { def self.fake_method; end })
```

Comment:
when I see this, what also about having as optional argument list of methods to fake? so it can be something like `self.define_missing_yast_module(name, methods: [:fake_method], &block)`

Also block variant looks kind of strange for me. I would expect that return value of block can be assigned to that name or even block invoked in context of that new Class. which would be the best from my POV.

So code like
```
Object.const_set(name.to_sym, Class.new do
  methods.each do { |m| define_singleton_method(m) {} }
 instance_eval(&block)
end)
```

This way. you can have for you Language example
```
Yast::RSpec::Helpers.define_yast_module("Language", methods: [:language])
```
or if you need specific execution in method
```
Yast::RSpec::Helpers.define_yast_module("Language") do
  def language
     "en_US"
  end
end
```

and with include I mention above, you can omit whole `Yast::RSpec::Helpers` part.